### PR TITLE
Improve resize handle management

### PR DESCRIPTION
**Pull request by mbaeuerle**
* Resize handles are redesigned and look more like in other applications
* Resize handles only appear on selected rectangles
* Visual handle is now independent of draggable handle area allowing for resizing also outside of drawn handle rectangle

Currently:
image
With this PR:
image

### CODE REVIEW

File: src/main/java/at/laborg/briss/gui/DrawableCropRect.java [hunk 1]
```diff
@@ -50,37 +63,86 @@ public DrawableCropRect(int x, int y, int width, int height) {
...
-        cornerLowerRight.setLocation(x + width - halfCornerDimension, y + height - halfCornerDimension);
+        cornerUpperLeft.setLocation(x, y);
+        cornerUpperRight.setLocation(x + width, y);
+        cornerLowerLeft.setLocation(x, y + height);
+        cornerLowerRight.setLocation(x + width, y + height);
     }
 
-    public void draw(Graphics2D g2) {
+    public boolean hasEnoughSpaceForHandles() {
+        return !(getWidth() < 2 * DrawableCropRect.CORNER_DIMENSION)
```
Comment (cleydyr) [inline src/main/java/at/laborg/briss/gui/DrawableCropRect.java:80]: I'd replace `!(a < b)` with `a >= b`.

File: src/main/java/at/laborg/briss/gui/DrawableCropRect.java [hunk 2]
```diff
@@ -50,37 +63,86 @@ public DrawableCropRect(int x, int y, int width, int height) {
...
+    }
+
+    private Font scaleFont(String text, FontMetrics fontMetrics) {
+
+        int size = MergedPanel.BASE_FONT.getSize();
+        float width = fontMetrics.stringWidth(text);
+        float height = fontMetrics.getHeight();
+        if (width == 0 || height == 0)
+            return MergedPanel.BASE_FONT;
+        float scaleFactorWidth = (float) (this.width * 0.95 / width);
```
Comment (cleydyr) [inline src/main/java/at/laborg/briss/gui/DrawableCropRect.java:91]: Replace magic constant 0.95 with named constant to reveal intent.

Comment (cleydyr) [reply src/main/java/at/laborg/briss/gui/DrawableCropRect.java:91]: Also, using `0.95f` would eliminate the need for the cast and improve readability.

File: src/main/java/at/laborg/briss/gui/DrawableCropRect.java [hunk 3]
```diff
@@ -50,37 +63,86 @@ public DrawableCropRect(int x, int y, int width, int height) {
...
+
+    private void drawSelectionOverlay(Graphics2D g2, FontMetrics fontMetrics) {
+        g2.setComposite(XOR_COMPOSITE);
+        g2.setColor(Color.BLACK);
+
+        g2.setStroke(SELECTED_STROKE);
+        g2.draw(this);
+
+        // display crop size in millimeters
+        int w = Math.round(25.4f * this.width / 72f);
```
Comment (cleydyr) [inline src/main/java/at/laborg/briss/gui/DrawableCropRect.java:137]: Avoid magic constants here as well.

File: src/main/java/at/laborg/briss/gui/DrawableCropRect.java [hunk 4]
```diff
@@ -143,4 +205,32 @@ public void setSize(int width, int height) {
...
+
+        public void draw(Graphics2D g2) {
+            g2.setColor(Color.WHITE);
+            g2.fillRect(corner.x - CORNER_DIMENSION / 2, corner.y - CORNER_DIMENSION /2, CORNER_DIMENSION, CORNER_DIMENSION);
+            g2.setColor(Color.BLACK);
+            g2.drawRect(corner.x - CORNER_DIMENSION / 2, corner.y - CORNER_DIMENSION /2, CORNER_DIMENSION, CORNER_DIMENSION);
+        }
+
+        public boolean contains(Point p) {
+            return p.x > corner.x - SELECTABLE_CORNER_DIMENSION / 2  && p.x < corner.x + SELECTABLE_CORNER_DIMENSION / 2 && p.y > corner.y - SELECTABLE_CORNER_DIMENSION && p.y < corner.y + SELECTABLE_CORNER_DIMENSION / 2;
```
Comment (cleydyr) [inline src/main/java/at/laborg/briss/gui/DrawableCropRect.java:233]: Use `Math.abs` here to save `&&`s:
```java
p.x > corner.x - SELECTABLE_CORNER_DIMENSION / 2 && p.x < corner.x + SELECTABLE_CORNER_DIMENSION / 2 == Math.abs(p.x - corner.x) < SELECTABLE_CORNER_DIMENSION / 2
```

File: src/main/java/at/laborg/briss/gui/DrawableCropRect.java (hunk 3 above)
Comment (mbaeuerle) [reply src/main/java/at/laborg/briss/gui/DrawableCropRect.java:137]: Actually I just moved the these magic constants just from the panel but I tried to come up with meaningful names provided I have understood them correctly 😄

**Comment by cleydyr** [comment]
Looks good to me. Merging. Thanks!

**Pull request merged**